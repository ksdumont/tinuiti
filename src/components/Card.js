import React from 'react'


function Card({product}) {
  const {category, dailyRankings, keyword, price, productDescription, productName, reviews, image} = product
  
  const displayStars = (rating) => {
    let node = document.createElement("div")
    for (let i = 0; i <= 5; i++) {
      if (i <= rating) {
        node.appendChild('<span class="fa fa-star checked"></span>')
      } else {
        node.appendChild('<span class="fa fa-star"></span>')
      }
    }
    return node;
  }

  return (
    <div className="card">
      <div className="card-img"><img src={image} alt={productName}/></div>
      <div className="card-body">
        <h5 className="card-title">{productName}</h5>
        <p className="card-price">{price}</p>
      <div className="card-body">
        <h5 className="card-rating">{reviews[0].rating}</h5>
        <p className="card-reviews">{reviews[0].count} reviews</p>
      </div>
      <button className="card-button">See More</button>
      </div>
    </div>
  )
}

export default Card
