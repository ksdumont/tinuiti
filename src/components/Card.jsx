import React from 'react';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import generateStars from '../helper';

function Card({ product }) {
  const {
    price,
    productId,
    productName,
    reviews,
  } = product;

  const history = useHistory();

  const handleClick = () => {
    history.push(`/product/${productId}`);
  };

  return (
    <div className="card">
      <div className="card-img"><img src="/lineGraphImage.jpg" alt="graph" /></div>
      <div className="card-body">
        <h5 className="card-title"><strong>{productName}</strong></h5>
        <p className="card-price">{price}</p>
        <h5 className="card-rating">
          {generateStars(reviews[0].rating)}
        </h5>
        <p className="card-reviews">
          {reviews[0].count}
          {' '}
          reviews
        </p>
        <button type="button" onClick={() => handleClick()} className="card-button">See More</button>
      </div>
    </div>
  );
}

Card.propTypes = {
  product: PropTypes.shape({
    image: PropTypes.string,
    price: PropTypes.string,
    productId: PropTypes.string,
    productName: PropTypes.string,
    reviews: PropTypes.instanceOf(Array),
  }),
};
Card.defaultProps = {
  product: PropTypes.shape({
    image: '/lineGraphImage.jpg',
    price: '$91.22',
    productId: '5f95f6674cf51d5b5197ab6f',
    productName: 'Brainclip',
    reviews: [{
      rating: 3,
      count: 903,
    }],
  }),

};

export default Card;
