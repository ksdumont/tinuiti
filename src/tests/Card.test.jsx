import React from 'react';
import ReactDOM from 'react-dom';
import Card from '../components/Card';
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import jsonData from '../data.json';

Enzyme.configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const wrapper = shallow(<Card product={jsonData[0]} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
