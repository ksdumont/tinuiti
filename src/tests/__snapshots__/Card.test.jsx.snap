// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-img"
  >
    <img
      alt="graph"
      src="/lineGraphImage.jpg"
    />
  </div>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      <strong>
        Brainclip
      </strong>
    </h5>
    <p
      className="card-price"
    >
      $91.22
    </p>
    <h5
      className="card-rating"
    >
      <i
        className="fa fa-star star-checked"
        key="1"
      />
      <i
        className="fa fa-star star-checked"
        key="2"
      />
      <i
        className="fa fa-star star-checked"
        key="3"
      />
      <i
        className="fa fa-star-o star-empty"
        key="4"
      />
      <i
        className="fa fa-star-o star-empty"
        key="5"
      />
    </h5>
    <p
      className="card-reviews"
    >
      903
       
      reviews
    </p>
    <button
      className="card-button"
      onClick={[Function]}
      type="button"
    >
      See More
    </button>
  </div>
</div>
`;
